### ISSUE TITLE
Add under maintenance page to Vortex

### ISSUE THREAD

**Issue by prayagd** [pendulum-chain/vortex#315]
As a vortex user, when vortex is not working i should be able to see the Under Maintenance page

## Acceptance criteria
* If Vortex is not working, this page should be displayed
* Copy of the page
    * Under Maintenance
    * We'll be back soon. 
* Reference
<img width="643" alt="Screenshot 2024-12-06 at 11 42 01 AM" src="https://github.com/user-attachments/assets/68271c14-1891-4b14-a83f-4f962b031403">

**Comment by prayagd** [comment]
@pendulum-chain/devs How would this work? Would we have to manually every time make this live?

**Comment by ebma** [comment]
If we don't want to make a new release every time something is not working, we should change the app so that it checks some endpoint of a status page (so not on the signer service but somewhere else) and if that status says that there is an error, vortex will show the under maintenance screen. We just need to find a provider that allows us to define custom statuses so that we can easily set it with a user interface (ie. you wouldn't need to require a dev to set the status).

**Comment by gianfra-t** [comment]
>  we should change the app so that it checks some endpoint of a status page

You mean something manual? We could set an endpoint on the backend that forces the status to be offline, and only check 1 endpoint from the UI which we already do. Not great UI for us, but works 😆.
But what do we want to shut it off? Ideally we have the pre-flight checks, are there cases not covered?

**Comment by ebma** [comment]
> We could set an endpoint on the backend that forces the status to be offline, and only check 1 endpoint from the UI which we already do. Not great UI for us, but works 😆.

True, but then we'd need to worry about the authentication, which I was hoping to offboard to some other provider.. :( we can't risk the endpoint being callable from anyone obviously. 

> But what do we want to shut it off? Ideally we have the pre-flight checks, are there cases not covered?

Generally yes, but for example the multicall contract migration that the squidrouter team did broke our app and there was some delay to get the fix live. I think for 'unforeseen' issues like this one, having the ability to show the maintenance message would be helpful.

**Comment by gianfra-t** [comment]
You're right, I didn't think about the authentication. That is inconvenient to add on our backend. Let's find something like you commented above.

**Comment by vadaynujra** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Pendulum-6408a0365cda59c57d39bee5/issues/pendulum-chain/vortex/315?planning-poker) with Zenhub @ebma @gianfra-t @Sharqiewicz

**Comment by ebma** [comment]
Instead of using another service we might be able to pull this off with GTM. 

One approach would be to have a custom trigger to add a cookie or similar to the app, which then indicates the 'under maintenance' mode. A quick list of actions generated by AI:
Is there a way I can toggle a flag in Google Tag Manager and then have access to that flags status in the web page?
Yes, you can achieve this by using Google Tag Manager (GTM) to set a variable or flag that can be accessed by your web page. Here's a step-by-step guide on how to do this:

### Step 1: Create a Variable in GTM

- Open GTM and navigate to your container.
- Go to Variables in the left-hand menu.
- Click on "New" to create a new user-defined variable.
- Name the variable (e.g., maintenanceMode).
- Set the variable type to 1st Party Cookie or another relevant type, depending on how you want to store the flag.
- Save the variable.

### Step 2: Create a Tag to Set the Flag

- Go to Tags in the left-hand menu.
- Click on "New" to create a new tag.
- Choose "Custom HTML" as the tag type.
- Add the HTML and JavaScript code to set the variable or flag. For example, if using cookies:

```html
<script>
    document.cookie = "maintenanceMode=true; path=/";
</script>
```

- Set up the trigger: Create a trigger that determines when this tag should fire (e.g., a button click, a specific page view, etc.).
- Save the tag.

### Step 3: Access the cookie
- Access the cookie and check it's value
- Show app content accordingly

**Comment by prayagd** [comment]
@ebma so tl;dr if i understand this right, we should manually set the variable to true or false and they the page shows up automatically right?

**Comment by ebma** [comment]
Yeah exactly. So we would configure Google Tag Manager and set up some tags so that in the vortex frontend, we have something we can check, maybe a cookie. And if the cookie is set to 'true' we assume that the app is 'under maintenance' and can then show the screen you put in the description. We would then in GTM configure the schedule of when this cookie should be set and it would be off by default of course.

**Comment by gianfra-t** [comment]
Interesting solution, so we rely on GTM setting of cookies. Do you know if the cookie will get removed automatically? Otherwise, we can overwrite it once the maintenance is over.

**Comment by ebma** [comment]
Apparently GTM triggers are quite customizable and they even allow defining a custom timespan for when it should be on/off. So we could e.g. say that a specific trigger, in this case the maintenance mode, should be fired from 19.12.2024 at 10am CET to 12am CET, or similar. You get the gist.

**Comment by prayagd** [comment]
@pendulum-chain/devs should i create a separate ticket for this? to do the above setup? What should the ticket be about?

**Comment by gianfra-t** [comment]
I think we can use this issue and make a PR for it.

**Comment by prayagd** [comment]
will add this to the epic list, IMO this is getting more and more important now
